### ISSUE TITLE
Customize jinja parser

### ISSUE THREAD

**Issue by fatal10110** [audreyr/cookiecutter#1088]
Hey guys, I'm trying to use the cookiecutter to generate a code base for my k8s + Helm project.
Helm uses "go templates" that has the same syntax like jinja templates, it causes conflicts, so I came up with a new feature... Jinja provides an ability to change it's syntax by modifying the environment.

We can provide these parameters as object in `cookiecutter.json`

```
    {
        "_environment": {
            "variable_start_string": "[[",
            "variable_end_string": "]]"
        }
    }
```

WDYT?

Here is the PR #1089

**Comment by fatal10110** [comment]
@matusf Nope, I can not... Because I need also cookiecutter tags there
example:

```
spec:
   container:
       image: {{ .Values.repository }}/{{ cookiecutter.image_name }}
```

**Comment by matusf** [comment]
@fatal10110,  this can be done with `raw statement` as well.
```
spec:
   container:
       image: {% raw %}{{ .Values.repository }}{% endraw %}/{{ cookiecutter.image_name }}
```
But I understand that it may not be the most elegant solution for every situation

**Comment by fatal10110** [comment]
Yeah, in big files that contains a lot of tags it may be a problem (almost hell :D)
https://github.com/helm/charts/blob/master/stable/acs-engine-autoscaler/templates/deployment.yaml

**Comment by senderle** [comment]
New cookiecutter user here, trying to understand the problem to avoid future issues like this. Aren't cookiecutter variables explicitly namespaced? I expected that when you render the template, cookiecutter would only touch the namespaced tags. As long as you've defined all the variables in your template, how do conflicts arise? Is this some kind of testing issue?

**Comment by fatal10110** [comment]
@senderle Nope, it uses template engine called "Jinja 2", so it doesn't really knows about cookiecutter...
It gets two inputs 1. Template string 2. Template params, so it will try to match any variables between `{{` and `}}`

**Comment by senderle** [comment]
@fatal10110 yeah, I use Django. Jinja is a clone of Django's template system. But this didn't seem to pose problems for [cookiecutter-django](https://github.com/pydanny/cookiecutter-django).

**Comment by senderle** [comment]
Though looking more closely at the code I see they are using 

    {% raw %} 

tags in some places. OK!

**Comment by ldaverio** [comment]
Hello everybody,

I would also like to be able to access the Jinja2 environment easily, typically to enable a line statement prefix such as `//#`, allowing me to write JS templates such as:

```javascript
//# if cookiecutter.mode == 'dev'
console.debug('options=' + JSON.stringify(options));
//# endif
```

**Comment by adzhurinskij** [comment]
Hi guys.

I cannot use cookiecutter for Ansible or Kubernetes, because it vary hard with `{{ raw }}` block. Code in files with this block is unreadable. Custom jinja configuration will solve the problem.

**Comment by con-f-use** [comment]
Just for completeness, this seems to solve the same problem as that #1227

**Comment by wkerzendorf** [comment]
I like this idea - is there any movement on this. Specifically for escapes in latex jinja2 templates.

**Comment by insspb** [comment]
@wkerzendorf something is already possible with master version. 
Check already merged https://github.com/cookiecutter/cookiecutter/pull/1339
